packages/flutter_driver/lib/src/driver/profiling_summarizer.dart

```diff
@@ -0,0 +1,154 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'percentile_utils.dart';
+import 'timeline.dart';
+
+/// The catrgory shared by all profiling related timeline events.
+const String kProfilingCategory = 'flutter::profiling';
+
+// These field names need to be in-sync with:
+// https://github.com/flutter/engine/blob/master/shell/profiling/sampling_profiler.cc
+const String _kCpuProfile = 'CpuUsage';
+const String _kGpuProfile = 'GpuUsage';
+const String _kMemoryProfile = 'MemoryUsage';
+
+/// Represents the supported profiling event types.
+enum ProfileType {
+  /// Profiling events corresponding to CPU usage.
+  CPU,
+
+  /// Profiling events corresponding to GPU usage.
+  GPU,
+
+  /// Profiling events corresponding to memory usage.
+  Memory,
+}
+
+/// Summarizes [TimelineEvents]s corresponding to [kProfilingCategory] category.
+///
+/// A sample event (some fields have been omitted for brewity):
+/// ```
+///     {
+///      "category": "flutter::profiling",
+///      "name": "CpuUsage",
+///      "ts": 121120,
+///      "args": {
+///        "total_cpu_usage": "20.5",
+///        "num_threads": "6"
+///      }
+///    },
+/// ```
+/// This class provides methods to compute the average and percentile information
+/// for supported profiles, i.e, CPU, Memory and GPU. Not all of these exist for
+/// all the platforms.
+class ProfilingSummarizer {
+  ProfilingSummarizer._(this.eventByType);
+
+  /// Creates a ProfilingSummarizer given the timeline events.
+  static ProfilingSummarizer fromEvents(List<TimelineEvent> profilingEvents) {
+    final Map<ProfileType, List<TimelineEvent>> eventByType =
```

Comment:
Nit: `eventsByType` to suggest that it's a list?

Reply:
Done